source/extensions/common/wasm/context.cc

```diff
@@ -1244,6 +1256,9 @@ void Context::log(const Http::RequestHeaderMap* request_headers,
                   const Http::ResponseHeaderMap* response_headers,
                   const Http::ResponseTrailerMap* response_trailers,
                   const StreamInfo::StreamInfo& stream_info) {
+  if (!in_vm_context_created_) {
+    return;
```

Comment:
Should access logger still log rejected request (...even when it didn't process it?)

Reply:
Humm... That interface is more independent.  However this is the onLog call which is called on a stream context.  There is a different type extension point which calls onLog on the Root context which is stateless.   That call will still  occur even if no other events have occurred. 

I would be tempted to return here because the user could have chosen the other extension point.  WDYT?

Reply:
SGTM, although it's a bit confusing that we have 2 separate extension points to do the same thing.

Reply:
I agree that it is a bit confusing, but this was a request from Mandar et al.   We can revisit it.